pkg/globalmanager/controllers/incrementallearning/downstream.go

```diff
@@ -62,18 +62,15 @@ func (c *Controller) syncToEdge(eventType watch.EventType, obj interface{}) erro
 	// more details at https://github.com/kubernetes/kubernetes/issues/3030
 	job.Kind = KindName
 
-	jobConditions := job.Status.Conditions
-	if len(jobConditions) == 0 {
-		return nil
-	}
-
 	dataName := job.Spec.Dataset.Name
+	// LC has dataset object on this node that may call dataset node
+	var dsNodeName string
 	ds, err := c.client.Datasets(job.Namespace).Get(context.TODO(), dataName, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("dataset(%s/%s) not found", job.Namespace, dataName)
+		klog.Errorf("not found job(name=%s/%s)'s dataset, error: %v", job.Kind, job.Name, err)
```

Comment:
There is no return here, maybe dsNodeName will be empty in the function below:
```
	if eventType == watch.Deleted {
		// delete jobs from all LCs
		nodes := sets.NewString(dsNodeName, trainNodeName, evalNodeName, deployNodeName)
		for node := range nodes {
			c.sendToEdgeFunc(node, eventType, job)
		}
		return nil
	}
```
`dsNodeName` in `sets.NewString(dsNodeName, trainNodeName, evalNodeName, deployNodeName)` maybe an empty string.

Reply:
the func `sendToEdgeFunc` will handle the case where the node is empty.